test/protocol/ConfigHandlerTest.js

```diff
@@ -951,6 +954,98 @@ describe("IBosonConfigHandler", function () {
           });
         });
       });
+
+      context("👉 setProtocolFeeTable()", async function () {
+        const FIVE_PERCENT = 500;
+        const TEN_PERCENT = 1000;
+        const TWENTY_PERCENT = 2000;
+
+        const feePriceRanges = [
+          parseUnits("1", "ether").toString(),
+          parseUnits("2", "ether").toString(),
+          parseUnits("5", "ether").toString(),
+        ];
+        const feePercentages = [FIVE_PERCENT, TEN_PERCENT, TWENTY_PERCENT];
+        let exchangeAmount, feeTier;
+        let expectedFeeAmount;
+        let usdcAddress;
+        let bosonToken;
+        beforeEach(async function () {
+          usdcAddress = await usdc.getAddress();
+          bosonToken = await token.getAddress();
+        });
+
+        it("should emit a FeeTableUpdated event", async function () {
+          // Set new protocol fee precentage address, testing for the event
+          await expect(configHandler.connect(deployer).setProtocolFeeTable(usdcAddress, feePriceRanges, feePercentages))
+            .to.emit(configHandler, "FeeTableUpdated")
+            .withArgs(usdcAddress, feePriceRanges, feePercentages, await deployer.getAddress());
+        });
+
+        it("should update state and return fee amount based on configured fee table", async function () {
+          await configHandler.connect(deployer).setProtocolFeeTable(usdcAddress, feePriceRanges, feePercentages);
+          for (let i = 0; i < feePriceRanges.length; i++) {
+            exchangeAmount = feePriceRanges[i];
+            feeTier = feePercentages[i];
+            expectedFeeAmount = applyPercentage(exchangeAmount, feeTier);
+            expect(await configHandler.getProtocolFee(usdcAddress, exchangeAmount)).to.equal(expectedFeeAmount);
+            expect(await configHandler.getProtocolFeePercentage(usdcAddress, exchangeAmount)).to.equal(feeTier);
+          }
+
+          // check for a way bigger value
+          feeTier = feePercentages[feePercentages.length - 1];
+          exchangeAmount = BigInt(feePriceRanges[feePriceRanges.length - 1]) * BigInt(2);
+          expectedFeeAmount = applyPercentage(exchangeAmount, feeTier);
+          expect(await configHandler.getProtocolFee(usdcAddress, exchangeAmount)).to.equal(expectedFeeAmount);
+          expect(await configHandler.getProtocolFeePercentage(usdcAddress, exchangeAmount)).to.equal(feeTier);
+        });
+
+        it("should update state and return boson flat fee if boson token used as exchange token", async function () {
+          await configHandler.connect(deployer).setProtocolFeeTable(usdcAddress, feePriceRanges, feePercentages);
```

Comment:
I have verified that with 1 await as is, the tx is confirmed and the protocol table has been set sucessfully

Reply:
Only a matter of luck because the hardhat node network is configured to mine immediately. It wouldn't work otherwise (or on a real blockchain).
It's working as is right now and not directly related to the PR, so ok to let it